### PR TITLE
Broken edition when no edition exists on url

### LINKED ISSUE #458: Broken edition when no edition exists on url.
Broken edition router after #455 when no edition with the id exists. The problem is caused by the edition not existing and the server returning an error or no edition because of this. It can be fixed by making sure undefined is returned when the edition does not exist.

### PR DESCRIPTION
**Pull request by jitsedesmet**
This PR closes #458.

### CODE REVIEW

File: frontend/src/components/util/routeInjector.tsx [hunk 1]
```diff
@@ -7,47 +7,74 @@ import { getAllEditionsFromPage, getEditionByName, getEditionOnUrl } from "../..
...
-        !contextEditionUrl && !routerEditionName
+        !contextEditionUrl && !routerEditionName && userIsLoggedIn
             ? getQueryUrlFromParams(apiPaths.editions, { sort: "year" })
             : null,
         getAllEditionsFromPage
     );
     const latestEdition = availableEditions?.at(0);
     const latestEditionName = latestEdition?.name;
 
+    console.log(contextEditionError || fetchedEditionError || availableEditionsError);
```
Comment (ruvmello) [inline frontend/src/components/util/routeInjector.tsx:47]: This always gets logged to the console, so this should probably go

File: frontend/src/components/util/routeInjector.tsx [hunk 2]
```diff
@@ -57,10 +84,18 @@ export default function RouteInjector({ children }: any) {
...
-    }, [fetchedRouterEditionUrl, contextEditionName, contextEditionUrl, latestEditionName]);
+    }, [
+        fetchedRouterEditionUrl,
+        hadRouterError,
+        contextEditionName,
+        contextEditionUrl,
+        hadContextError,
+        latestEditionName,
+    ]);
+    // console.log([fetchedRouterEditionUrl, hadRouterError, contextEditionName, contextEditionUrl, hadContextError, latestEditionName])
```
Comment (ruvmello) [inline frontend/src/components/util/routeInjector.tsx:95]: Can go

**Review by ruvmello — CHANGES_REQUESTED** [review]

File: frontend/src/components/util/routeInjector.tsx (hunk 1 above)
Comment (jitsedesmet) [reply frontend/src/components/util/routeInjector.tsx:47]: Def!

File: frontend/src/components/util/routeInjector.tsx (hunk 2 above)
Comment (jitsedesmet) [reply frontend/src/components/util/routeInjector.tsx:95]: Def!

**Review by ruvmello — APPROVED** [review]

**Pull request merged**